tests/unit_tests/db_engine_specs/test_trino.py

```diff
@@ -366,3 +369,37 @@ def test_handle_cursor_early_cancel(
         assert cancel_query_mock.call_args[1]["cancel_query_id"] == query_id
     else:
         assert cancel_query_mock.call_args is None
+
+
+@pytest.mark.parametrize(
+    "data,description,expected_result",
+    [
+        (
+            [["1.23456", "abc"]],
+            [("dec", "decimal(12,6)"), ("str", "varchar(3)")],
+            [(Decimal("1.23456"), "abc")],
```

Comment:
Notice how we're changing from `list[list[Any]]` to `list[tuple[Any, ...]]` here - this is because this is what SQLA drivers *should* return. This will save us from having to do the conversion later here: https://github.com/apache/superset/blob/50535e427e51743e298d46009d51b08ef2380012/superset/result_set.py#L132-L134